src/event/PluginInterface.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020-2023 Delft University of Technology and SynTest contributors
+ *
+ * This file is part of SynTest Framework - SynTest Core.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/no-empty-function */
+
+import { Encoding } from "..";
+import { ProgramState } from "./ProgramState";
+
+/**
+ * The PluginInterface is an interface for creating plugins for the core of the Syntest Framework.
+ *
+ * Format of functions:
+ * on[EVENT]Start
+ * on[EVENT]Complete
+ */
+export abstract class PluginInterface<T extends Encoding> {
```

Comment:
Should we not call this listener interface given that we register listeners and not all listeners necessarily have to be plugins

Reply:
Good point.